test-loop-tests/src/tests/catching_up.rs

```diff
@@ -0,0 +1,515 @@
+                    transaction: tx.clone(),
+                    is_forwarded: false,
+                    check_only: false,
+                });
+            }
+            tx_count += 1;
+        }
+    }
+    assert_eq!(tx_count, 16 * 16);
+
+    runner.run_for_an_epoch(&mut env);
+    runner.run_for_an_epoch(&mut env);
+
+    let mut amounts = HashMap::new();
+
+    assert_eq!(env.node_datas.len(), 16);
+    let view_client_handlers =
+        env.node_datas.iter().map(|data| data.view_client_sender.actor_handle()).collect_vec();
+
+    let client = &env.test_loop.data.get(client_actor_handle).client;
+    let head = client.chain.head().unwrap();
+    let start_height = head.height;
+    let target_height = epoch_length * 6 + 2;
+    runner.run_until(
+        &mut env,
+        |test_loop_data| {
+            for handler in &view_client_handlers {
+                let view_client = test_loop_data.get_mut(handler);
+                for test_account in &test_accounts {
+                    let Ok(response) = view_client.handle(Query::new(
+                        BlockReference::latest(),
+                        QueryRequest::ViewAccount { account_id: test_account.clone() },
+                    )) else {
+                        continue;
+                    };
+                    if let QueryResponseKind::ViewAccount(view_account_result) = response.kind {
+                        let amount = view_account_result.amount;
+                        match amounts.entry(test_account.clone()) {
+                            Entry::Occupied(entry) => {
+                                assert_eq!(
+                                    *entry.get(),
+                                    amount,
+                                    "OCCUPIED {entry:?}; value != {amount}"
+                                );
+                            }
+                            Entry::Vacant(entry) => {
+                                println!("VACANT {:?}", entry);
+                                if non_zero {
+                                    assert_ne!(amount % 100, 0);
+                                } else {
+                                    assert_eq!(amount % 100, 0);
+                                }
+                                entry.insert(amount);
+                            }
+                        }
+                    }
+                }
+            }
+            let client = &test_loop_data.get(client_actor_handle).client;
+            let head = client.chain.head().unwrap();
+            head.height >= target_height
+        },
+        block_prod_time * (target_height - start_height) as i32,
+    );
+
+    for test_account in &test_accounts {
+        match amounts.entry(test_account.clone()) {
+            Entry::Occupied(_) => {
+                continue;
+            }
+            Entry::Vacant(entry) => {
+                println!("VALIDATOR = {:?}, ENTRY = {:?}", test_account, entry);
```

Comment:
nit: use tracing and make it into an error log (tracing::error!)

Reply:
Are there benefits to doing so? IIUC these would be useful only for someone debugging the test at which point they would have to read it in any case and would know where print statements come from. They are also less verbose so may be a bit easier to read compared to tracing. Additionally not all prints indicate errors, I think this one does not.

Reply:
It's a convention to use tracing for logging but that's about it.

Reply:
For non-errors you can use tracing::info or tracing::debug